modules/apps/polls/polls-test/src/testIntegration/java/com/liferay/polls/search/test/PollsQuestionIndexerIndexedFieldsTest.java

```diff
@@ -0,0 +1,225 @@
+
+	@Before
+	public void setUp() throws Exception {
+		setUpUserSearchFixture();
+
+		setUpIndexedFieldsFixture();
+
+		setUpPollsQuestionFixture();
+
+		setUpPollsQuestionIndexerFixture();
+	}
+
+	@Test
+	public void testIndexedFields() throws Exception {
+		PollsQuestion pollsQuestion =
+			pollsQuestionFixture.createPollsQuestion();
+
+		String searchTerm = pollsQuestion.getDescription(Locale.US);
+
+		Document document = pollsQuestionIndexerFixture.searchOnlyOne(
+			searchTerm);
+
+		indexedFieldsFixture.postProcessDocument(document);
+
+		FieldValuesAssert.assertFieldValues(
+			_expectedFieldValues(pollsQuestion), document, searchTerm);
+	}
+
+	protected void setUpIndexedFieldsFixture() {
+		indexedFieldsFixture = new IndexedFieldsFixture(
+			resourcePermissionLocalService, searchEngineHelper);
+	}
+
+	protected void setUpPollsQuestionFixture() {
+		pollsQuestionFixture = new PollsQuestionFixture(_group);
+
+		_pollsQuestions = pollsQuestionFixture.getPollsQuestions();
+	}
+
+	protected void setUpPollsQuestionIndexerFixture() {
+		pollsQuestionIndexerFixture = new IndexerFixture<>(PollsQuestion.class);
+	}
+
+	protected void setUpUserSearchFixture() throws Exception {
+		userSearchFixture = new UserSearchFixture();
+
+		userSearchFixture.setUp();
+
+		_group = userSearchFixture.addGroup();
+
+		_groups = userSearchFixture.getGroups();
+
+		_users = userSearchFixture.getUsers();
+	}
+
+	protected IndexedFieldsFixture indexedFieldsFixture;
+	protected PollsQuestionFixture pollsQuestionFixture;
+	protected IndexerFixture<PollsQuestion> pollsQuestionIndexerFixture;
+
+	@Inject
+	protected ResourcePermissionLocalService resourcePermissionLocalService;
+
+	@Inject
+	protected SearchEngineHelper searchEngineHelper;
+
+	protected UserSearchFixture userSearchFixture;
+
+	private Map<String, String> _expectedFieldValues(
+			PollsQuestion pollsQuestion)
+		throws Exception {
+
+		Map<String, String> map = new HashMap<>();
+
+		map.put(Field.COMPANY_ID, String.valueOf(pollsQuestion.getCompanyId()));
+		map.put(Field.DESCRIPTION, _getDescriptionField(pollsQuestion));
+		map.put(
+			Field.ENTRY_CLASS_PK,
+			String.valueOf(pollsQuestion.getQuestionId()));
+		map.put(Field.ENTRY_CLASS_NAME, PollsQuestion.class.getName());
+		map.put(Field.GROUP_ID, String.valueOf(pollsQuestion.getGroupId()));
+		map.put(
+			Field.SCOPE_GROUP_ID, String.valueOf(pollsQuestion.getGroupId()));
+		map.put(Field.STAGING_GROUP, String.valueOf(_group.isStagingGroup()));
+		map.put(Field.TITLE, _getTitleField(pollsQuestion));
+		map.put(Field.USER_ID, String.valueOf(pollsQuestion.getUserId()));
+		map.put(
+			Field.USER_NAME, StringUtil.lowerCase(pollsQuestion.getUserName()));
+		map.put(
+			"title_sortable",
+			StringUtil.lowerCase(_getTitleField(pollsQuestion)));
+
+		indexedFieldsFixture.populateUID(
+			PollsQuestion.class.getName(), pollsQuestion.getQuestionId(), map);
+
+		_populateDates(pollsQuestion, map);
+
+		_populateRoles(pollsQuestion, map);
```

Comment:
@brandizzi In the new Indexer "PollsQuestion" has the fields "roleId" and "groupRoleId" because a "ModelResourcePermission" was registered in "PollsQuestionModelResourcePermissionRegistrar"